packages/ibm-products/src/global/js/hooks/usePresence.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect, useState, RefObject } from 'react';
+
+// Used as a replacement for AnimatePresence from framer-motion,
+// this hook will return a boolean value to let the component
+// calling this hook to know whether it should render or not
+export const usePresence = (
+  open: boolean,
+  ref: RefObject<HTMLElement>,
+  animationName: string
```

Comment:
I included the `animationName` as a required parameter so that this hook can be sure to check against `event.animationName` to be sure it's the animation we're anticipating before changing `shouldRender` (it could be possible an element has multiple `animationend` events triggered via different classes).